### PR TITLE
Student result

### PR DESCRIPTION
**Pull request by parthshah51999**

### CODE REVIEW

File: src/BreadCrumbs.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const crumbs = ['Home', 'StudentList', 'DetailResult'];
+
+export default class BreadCrumbs extends React.Component {
+  breadCrumbClick(page) {
+    return this.props.breadCrumbClick(page);
+  }
+
+  render() {
+    let fCrumbs = [...crumbs];
+    this.props.currentPage != 'DetailResult' ? fCrumbs = crumbs.slice(0, 2) : fCrumbs = crumbs;
```
Comment (visha12l) [inline src/BreadCrumbs.js]: fCrumbs = this.props.currentPage != 'DetailResult' ?  crumbs.slice(0, 2) : crumbs;

File: src/DetailResult.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export default class DetailResult extends React.Component {
+  render() {
+    const indStudent = this.props.indStudent;
```
Comment (visha12l) [inline src/DetailResult.js]: use object destructuring

File: src/DetailResult.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    return (
+      <div className="Result">
+        <div><b>Name English Hindi Mathematics</b></div>
+        <div>{indStudent.firstName} {indStudent.lastName} {indStudent.marks.english} {indStudent.marks.hindi} {indStudent.marks.mathematics}</div>
```
Comment (visha12l) [inline src/DetailResult.js]: use object destructuring

File: src/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default class HomePage extends React.Component {
```
Comment (visha12l) [inline src/HomePage.js]: make it as a stateless component

File: src/ResultPage.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  render() {
+    return (
+      <div>
+        Click on the person to get the detailed result.
+        <ul><b>FirstName LastName Percentage </b></ul>
+        {
+          this.props.studentsData.map((student, i) => {
+            let total = (student.marks.english + student.marks.hindi + student.marks.mathematics);
```
Comment (visha12l) [inline src/ResultPage.js]: use only one let 
use object destructuring

File: src/main.js [hunk 6]
```diff
@@ -1,9 +1,57 @@
...
+
+  breadCrumbClick(displayPage) {
+    this.setState({
+      displayPage,
+    })
+  }
+
+  displayResult(displayPage, indStudent) {
+    this.setState({
+      displayPage,
```
Comment (visha12l) [inline src/main.js]: good use of enhanced object literals :+1:

File: src/main.js [hunk 7]
```diff
@@ -1,9 +1,57 @@
...
+      indStudent,
+    })
+  }
+
+  doRenderPageContent() {
+    if(this.state.displayPage === 'Home')
```
Comment (visha12l) [inline src/main.js]: space after if

Comment (visha12l) [inline src/main.js]: use object destructuring

File: src/main.js [hunk 8]
```diff
@@ -1,9 +1,57 @@
...
+    else if (this.state.displayPage === 'DetailResult')
+      return <DetailResult indStudent={this.state.indStudent}/>
+  }
+
   render() {
     return (
-      <div className="parent"> React Boiler Plate </div>
+      <div className="PageContent">
+        <BreadCrumbs
+        breadCrumbClick={this.breadCrumbClick}
```
Comment (visha12l) [inline src/main.js]: indentation

File: src/BreadCrumbs.js (hunk 1 above)
Comment (parthshah51999) [reply src/BreadCrumbs.js]: resolved in eslint correction.

File: src/ResultPage.js (hunk 5 above)
Comment (parthshah51999) [reply src/ResultPage.js]: resolved in eslint correction.

File: src/main.js (hunk 6 above)
Comment (parthshah51999) [reply src/main.js:29]: :)

File: src/main.js (hunk 7 above)
Comment (parthshah51999) [reply src/main.js]: resolved in eslint correction.

File: src/main.js (hunk 8 above)
Comment (parthshah51999) [reply src/main.js]: resolved in eslint correction.

File: src/DetailResult.js (hunk 2 above)
Comment (parthshah51999) [reply src/DetailResult.js]: resolved in eslint correction.

File: src/HomePage.js (hunk 4 above)
Comment (parthshah51999) [reply src/HomePage.js]: resolved in eslint correction.

File: src/main.js (hunk 7 above)
Comment (parthshah51999) [reply src/main.js]: resolved in eslint correction.

**Pull request merged**